### ISSUE TITLE
Linting breaks with value imports

### ISSUE THREAD

**Issue by Moncefmd** [helm/helm#3495]
## Helm version used : 2.8.1

Linting basically breaks when importing dictionaries from a child chart, even though helm resolves the import correctly at the end (the chart installs with the correct values). Here is a simple example on how to reproduce the issue :

Suppose we have a parent chart and a child chart from which we import some values.

Child's values.yaml :
```
exports:
  data:
    key1: "value1"
    nestedkey: 
      key2: "value2"
```

In parent requirements.yaml :
```
    import-values:
      - data
```

Now when when I reference the first imported value in the parent's deploy.yaml `{{  .Values.key1 }}`, linting passes without errors. But when I reference the second one,  `{{  .Values.nestedkey.key2 }}`, the lint breaks :

> [ERROR] templates/: render error in "parent/templates/deploy.yaml": template: parent/templates/deploy.yaml:5:20: executing "parent/templates/deploy.yaml" at <.Values.nestedkey.my...>: can't evaluate field mykey in type interface {}
> 
> Error: 1 chart(s) linted, 1 chart(s) failed

Even though when installed or dry-runned, both imported values are resolved correctly.

**Comment by kitd** [comment]
I'm having a similar problem, except when the child values are made available under an alias. Eg:

`common-lib/values.yaml`:
```
profile:
  name: My Name
  version: 1234-5678
  created: 2018-02-22
```
Parent `requirements.yaml`:
```
dependencies:
  - name: common-lib
    repository: file://../common-lib
    version: 0.1.0
    alias: common
```
An excerpt from a parent manifest yaml:
```
...
  annotations:
{{ toYaml .Values.common.profile | indent 4 }}    
...
```
Running the template finds the values successfully and shows the computed `common` values in debug mode, but the linter fails with
```
[ERROR] templates/: render error in "lint-issue/templates/deployment.yaml": template: lint-issue/templates/deployment.yaml:11:17: 
executing "lint-issue/templates/deployment.yaml" at <.Values.common.profi...>: 
can't evaluate field profile in type interface {}
```

**Comment by luisdavim** [comment]
/remove-lifecycle rotten
/remove-lifecycle stale

I have the same issue.

**Comment by jnsvd** [comment]
I'm also experiencing this issue.

**Comment by zzarkouna** [comment]
i have the same error with helm3.0.0

**Comment by strainovic** [comment]
Same problem with Helm 3.2.4 when using `index` function over the aliased chart name to get value from the chart with the dash in its name.

`{{ index .Values "alias-chart-name" "key" }}`

`helm lint` returns `error calling index: index of nil pointer` while `helm template` and `helm install/upgrade` work fine.

Without alias, lint works as expected.

Any update on this topic?

**Comment by gfbett** [comment]
Same issue here
/remove-lifecycle rotten

**Comment by aramhakobyan** [comment]
Same issue here

**Comment by ns-jisorce** [comment]
Reopen ?

**Comment by than6oz** [comment]
Not sure if this issue has been resolved? I'm still having the issue with helm 3.6.3

**Comment by alexandre-girard-maif** [comment]
still having the same issue

**Comment by sunglim** [comment]
still having a problem with this issue

**Comment by Unknow0** [comment]
Hi,
i'm having the same issues in 3.7.1
Regards.

**Comment by RinckFox** [comment]
If someone came here, like me, looking for an answer - this bug seems to be fixed in v3.8.0.
